File: 2022/go/6.go
package main

import (
	"fmt"
	"os"
)

func main() {

	input, err := os.ReadFile("../6.txt")
	if err != nil {
		fmt.Printf("Could not read the file due to this %s error \n", err)
	}
	
	line := string(input)

	index := 0
	for i := 0; i < len(line); i++ {
		chunk := map[string]bool{}

		for j := i; j < i + 4; j++ {
			chunk[string(line[j])] = true
		}

		if len(chunk) == 4 {
			break
		}
		index += 1
	}
	//
	fmt.Printf("%v\n", index + 4)

	index = 0
	for i := 0; i < len(line); i++ {
		chunk := map[string]bool{}

		for j := i; j < i + 14; j++ {
			chunk[string(line[j])] = true
		}

		if len(chunk) == 14 {
			break
		}
		index += 1
	}

	fmt.Printf("%v\n", index + 14)

}

File: 2022/rust/6.rs
use std::fs;
use std::collections::HashSet;

fn main() {
    let input = fs::read_to_string("../6.txt").unwrap();
    let signal : Vec<char> = input.chars().collect();
  
    let size = signal.len();
    let mut index = 0;
    
    for it in signal.windows(4) {
        let chunk: HashSet<_> = it.iter().collect();
        if chunk.len() == 4 {
            break;
        }
        index += 1;
    }

    println!("Index : {}", index + 4);
    index = 0;
    for it in signal.windows(14) {
        let chunk: HashSet<_> = it.iter().collect();
        if chunk.len() ==  14 {
            break;
        }
        index += 1;
    }
    println!("Index : {}", index + 14);
}